File: app/src/main/java/me/pwcong/jpstart/mvp/presenter/JPStartTabFragmentPresenterImpl.java
package me.pwcong.jpstart.mvp.presenter;

import me.pwcong.jpstart.App;
import me.pwcong.jpstart.mvp.model.BaseModel;
import me.pwcong.jpstart.mvp.model.JPStartTabFragmentModelImpl;
import me.pwcong.jpstart.mvp.view.BaseView;

/**
 * Created by Pwcong on 2016/9/27.
 */

public class JPStartTabFragmentPresenterImpl extends BasePresenter<BaseView.JPStartTabFragmentView> implements BasePresenter.JPStartTabFragmentPresenter {

    private BaseModel.JPStartTabFragmentModel model;

    public JPStartTabFragmentPresenterImpl(BaseView.JPStartTabFragmentView view) {
        super(view);
        model = new JPStartTabFragmentModelImpl();
    }

    @Override
    public void initJPStartTabFragment() {
        view.setData(model.getData());
        view.scrollViewPager(App.CURRENT_ITEM);
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/mvp/presenter/impl/JPStartTabFragmentPresenterImpl.kt
package com.github.pwcong.jpstart.mvp.presenter.impl

import com.github.pwcong.jpstart.App
import com.github.pwcong.jpstart.mvp.model.BaseModel.JPStartTabFragmentModel
import com.github.pwcong.jpstart.mvp.model.impl.JPStartTabFragmentModelImpl
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter.JPStartTabFragmentPresenter
import com.github.pwcong.jpstart.mvp.view.BaseView.JPStartTabFragmentView

class JPStartTabFragmentPresenterImpl(view: JPStartTabFragmentView) :
    BasePresenter<JPStartTabFragmentView>(view), JPStartTabFragmentPresenter {
    private val model: JPStartTabFragmentModel = JPStartTabFragmentModelImpl()

    override fun initJPStartTabFragment() {
        view.setData(model.data)
        view.scrollViewPager(App.CURRENT_ITEM)
    }
}
